src/sdf/src/sdf.cpp

```diff
@@ -334,10 +334,16 @@ Mesh LevelSet(std::function<float(glm::vec3)> sdf, Box bounds, float edgeLength,
       const Uint64 lastMorton =
           MortonCode(glm::ivec4((lastVert - bounds.min) / spacing, 1));
       const float ratio = static_cast<float>(maxMorton) / lastMorton;
+
+      if (tableSize >= std::numeric_limits<int>::max())
+        throw std::out_of_range("mesh too large");
+
       if (ratio > 1000)  // do not trust the ratio if it is too large
         tableSize *= 2;
       else
         tableSize *= ratio;
+      if (tableSize > std::numeric_limits<int>::max())
+        tableSize = std::numeric_limits<int>::max();
```

Comment:
If we've changed it to index with `size_t`, then shouldn't we be able to handle a larger max length?

Reply:
Ah, this was because `used` is `uint32_t`. Thinking about it, we can just use a size_t, don't even need the vec.